lambdaclass/mirra_backend
# [GH-633] Add quest rewards functionality

## Motivation

Now that a player can have quests when we complete them we should award players with quests rewards defined im the config, for the time being, they'll be currencies

Closes #633

## Summary of changes

- Added quests completion on match registration

## How to test it?
Playing:
1. Import quests from file: `GameBackend.CurseOfMirra.Config.import_quest_descriptions_config` 
2. Go to the game client, register as a google client and copy your id
3. Get your google user `{:ok, google_user} = GameBackend.Users.get_google_user(google_user_id)`
4. Add quests to user: `quests = GameBackend.CurseOfMirra.Quests.add_quest_to_user_id(google_user.user.id, 3, "daily")`
5. Play a match and lose
6. If you head to the terminal an get your google user currencies you should see 444 gold and the quests marked as completed

Terminal:
1. Create a user: `{:ok, google_user} = GameBackend.Users.find_or_create_google_user_by_email("a")`
7. Import quests from file: `GameBackend.CurseOfMirra.Config.import_quest_descriptions_config`
8. Add quests to user: quests = GameBackend.CurseOfMirra.Quests.add_quest_to_user_id(google_user.user.id, 3, "daily")
9. insert a match result:
```
attrs = %{
    match_id: "f7c68e73-b797-428c-91aa-c2e2b119a17b",
    result: "loss",
    kills: 43,
    deaths: 0,
    character: "uma",
    position: 1,
    google_user_id: google_user.id,
    damage_done: 444,
    damage_taken: 0,
    health_healed: 444,
    killed_by_bot: false,
    duration_ms: 4444
}

GameBackend.Matches.ArenaMatchResult.changeset(%GameBackend.Matches.ArenaMatchResult{}, attrs) |> GameBackend.Repo.insert
```
10. Get the google user with the match results updated `[user] = GameBackend.Users.get_google_users_with_todays_daily_quests([google_user.id])`
11. Since we have the quests `Lose a match` you should have that one completed, if you repeat 4 again you should also complete the `Get 44 kills` quests, run this command to check completed quests `GameBackend.CurseOfMirra.Quests.get_google_user_daily_quests_completed(user)`
Disclaimer: This will only display the quests completed, it won't insert any currencies. To check that you can play a match 

## Checklist
